### DISCUSSION TITLE
Complex queries possible?

### DISCUSSION THREAD

**Discussion by cheema-corellian** [Azure/data-api-builder#1850] [Q&A]
Hello,

I found out about this project after watching GraphQL Seattle Oct '22 event video on YouTube.  This looks like a promising product and I am very interested in using it.  My initial question is about the possibility of running complex queries.  Watching the video, I got the impression that simple joins/relationships are possible.  But it is not clear if complex queries, e.g. "group by" are possible as well.  I frequently need to group my results by some field sometimes also need to count some values in a subquery.

If such things are possible, please let me know how to test them.  Thanks!

**Comment by yorek** [comment]
Hi @cheema-corellian! At the moment, queries involving aggregations or more advanced analytical functions (like windowing functions) are not available. The idea is that GraphQL is really not a complete data query language, so if you need to add aggregations and other more advanced data manipulation constructs, the best way to do it would be to create a view or a stored procedure and then call it via GraphQL.

Having said that, I'm extremely interested in the use cases that you have and that would require aggregations via GraphQL. In addition to that, would be great to know what kind of aggregations you need.  Thanks!
